r4627 | exg | 2007-10-22 13:27:56 +0000 (Mon, 22 Oct 2007) | 3 lines

Handle server == NULL case in skip_target, fix regression with irssi-icb.
Patch by Stuart Henderson, bug #539.

Index: trunk/src/fe-common/irc/fe-irc-messages.c
===================================================================
--- trunk/src/fe-common/irc/fe-irc-messages.c	(revision 4626)
+++ trunk/src/fe-common/irc/fe-irc-messages.c	(revision 4627)
@@ -44,7 +44,7 @@
 	const char *val, *chars;
 
 	/* Quick check */
-	if (server->prefix[(int)(unsigned char)*target] == 0)
+	if (server == NULL || server->prefix[(int)(unsigned char)*target] == 0)
 		return target;
 
 	/* Hack: for bahamut 1.4 which sends neither STATUSMSG nor
